### ISSUE TITLE
(Question)ibus-pinyin or ibus-libpinyin setup problem for stumpwm

### ISSUE THREAD

**Issue by wxie2017** [stumpwm/stumpwm#1131]
environment:
1. Debian GNU/Linux 12 - minimal installation without any desktop
2. apt install xorg, sbcl
3. apt install stumpwm
4. install ibus-pinyin

init files:
.bash_profile
# ibus input
  export GTK_IM_MODULE=ibus
  export XMODIFIERS=@im=ibus
  export QT_IM_MODULE=ibus
  ibus-daemon -drxR -v &

.xinitrc
#!/usr/bin/env bash

eval `gnome-keyring-daemon --start --components=gpg,pkcs11,secrets,ssh`
export GNOME_KEYRING_CONTROL
export SSH_AUTH_SOCK
export GPG_AGENT_INFO
export GNOME_KEYRING_PID
eval $(ssh-agent)

exec /usr/bin/stumpwm

setup:
im-config -c
use default

ibus-setup
input method: I see English - English (US); I click ADD; then the setup window freezes, and I can C-c to kill it.
The Chinese keyboard input method cannot be activated using s-SPC or any other methods.

What I miss in the process?

**Comment by dmb2** [comment]
Probably nothing? While lisp supports utf8 characters fairly well, stumpwm does not.  It handles everything as 8bit character strings and assumes their ASCII encoded.

This is usually an OK assumption since clx doesn't support the Xkeyboard extension to X11, and so the lisp X11 server cannot handle multibyte character requests.

I do not know the internals of ibus well enough to know where the pinyin input method fits into the X11 software stack.  It might be that stumpwm is choking on utf8 strings, or its just not responding to an X11 request that ibus is sending.

In short, its a complicated problem that we don't support.

**Comment by wxie2017** [comment]
Thanks.
I switch to fcitx. It works for Firefox windows which is enough for me.
